lib/matplotlib/axis.py

```diff
@@ -1096,10 +1096,20 @@ def _translate_tick_params(kw, reverse=False):
             'labeltop': 'label2On',
         }
         if reverse:
-            kwtrans = {
-                oldkey: kw_.pop(newkey)
-                for oldkey, newkey in keymap.items() if newkey in kw_
-            }
+            kwtrans = {}
+            is_x_axis = cls.axis_name == 'x'
+            for oldkey, newkey in keymap.items():
+                if newkey in kw_:
+                    if is_x_axis and newkey == 'label1On':
+                        kwtrans['labelbottom'] = kw_.pop(newkey)
+                    elif is_x_axis and newkey == 'tick1On':
+                        kwtrans['bottom'] = kw_.pop(newkey)
+                    elif is_x_axis and newkey == 'label2On':
+                        kwtrans['labeltop'] = kw_.pop(newkey)
+                    elif is_x_axis and newkey == 'tick2On':
+                        kwtrans['top'] = kw_.pop(newkey)
```

Comment:
```suggestion
            y_axis_keys = ['left', 'right', 'labelleft', 'labelright']
            for oldkey, newkey in keymap.items():
                if newkey in kw_:
                    if is_x_axis and oldkey in y_axis_keys:
                        continue
```
Is it worth simplifying here?  With the long if loop we are manually re-stating mappings that already exist in `kwtrans`.  I think we just want to skip if we are on the x-axis but the `oldkey` is specific to the y-axis.